src/Qir/Execution/Tests.Microsoft.Quantum.Qir.Tools/Util.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.IO;
+
+namespace Tests.Microsoft.Quantum.Qir.Tools
+{
+    internal static class Util
+    {
+        public static bool CompareFiles(FileInfo fileA, FileInfo fileB)
+        {
+            if (fileA.FullName == fileB.FullName)
+            {
+                return true;
+            }
+
+            using var fileStreamA = fileA.OpenRead();
+            using var fileStreamB = fileB.OpenRead();
+            return CompareStreams(fileStreamA, fileStreamB);
+        }
+
+        public static bool CompareStreams(Stream streamA, Stream streamB)
+        {
+            if (streamA.Length != streamB.Length)
+            {
+                return false;
+            }
+
+            (streamA.Position, streamB.Position) = (0, 0);
+            (int byteA, int byteB) = (0, 0);
+            while ((byteA == byteB) && (byteA != -1))
+            {
+                (byteA, byteB) = (streamA.ReadByte(), streamB.ReadByte());
```

Comment:
Probably good enough for us, but see discussion here on performance. https://stackoverflow.com/questions/43289/comparing-two-byte-arrays-in-net

Reply:
Yes, I believe for our use it should be good enough.